### DISCUSSION TITLE
TaskManagerIO and FreeRTOS example

### DISCUSSION THREAD

**Discussion by fletchjp** [TcMenu/TaskManagerIO#54] [General]
I have just found in another discussion thread the following:

_Many people when they first see task manager think that it is akin to FreeRTOS, but it is not. It is designed to co-exist nicely with FreeRTOS and provide a simplified solution for scheduling and events. Really, it was originally the core of tcMenu, but it was so useful that we broke it out._

I am interested in this and have begun to explore how to combine TaskManagerIO with FreeRTOS on both on Arduino MEGAs and DUEs (with different versions of FreeRTOS).

So far I have put the call to taskManager,runLoop() into a FreeRTOS task, with a vTaskDelay for 100 milliseconds.

Are there any examples available of how to do the combination?

Thanks

John

**Comment by davetcc** [comment]
Hi there, the example for mbed RTOS includes a threaded example:

examples/mbedRtos/mbedExample.cpp

**Reply by fletchjp** [reply]
Thank you, I have downloaded a copy and will convert it to an Arduino IDE example. John
